### PR TITLE
 Handle role change error in `cluster setslot` when migrating the last slot away with allow-replica-migration enabled

### LINKED ISSUE #463: 12-replica-migration-2 fails with `UNBLOCKED force unblock from blocking operation, instance state changed (master -> replica?)`
Does any of the CI runs do `make test-cluster`? We might want to enable cluster tests in at least one run @madolson 

I am able to consistently get a test failure after this commit. @PingXie can you please take a look?

```bash
./runtest-cluster --single tests/12-replica-migration-2
```

_Originally posted by @srgsanky in https://github.com/valkey-io/valkey/issues/445#issuecomment-2099625063_

### PR DESCRIPTION
**Pull request by PingXie**
Fix #463

### CODE REVIEW

**Comment by PingXie** [comment]
@srgsanky can you please review the fix? Thanks

**Comment by srgsanky** [comment]
Looks good, thanks!

File: src/valkey-cli.c [hunk 1]
```diff
@@ -5239,13 +5239,30 @@ static int clusterManagerMoveSlot(clusterManagerNode *source,
...
         success = clusterManagerSetSlot(source, target, slot, "node", err);
-        const char *acceptable = "ERR Please use SETSLOT only with masters.";
-        if (!success && err && !strncmp(*err, acceptable, strlen(acceptable))) {
-            zfree(*err);
-            *err = NULL;
-        } else if (!success && err) {
+        if (!success && err) {
+            const char *acceptable[] = {
+                "ERR Please use SETSLOT only with masters.",
+                "UNBLOCKED force unblock from blocking operation, instance state changed (master -> replica?)"};
```
Comment (madolson) [inline src/valkey-cli.c]: We tell end users not to take a dependency on the full string, just the code "UNBLOCKED". Should we do that here as well so we can change the rest of the language later?

Comment (PingXie) [reply src/valkey-cli.c]: That is a good point. Let me fix this one first but we will likely need to scrub valkey-cli.c too (in future PRs)

Comment (madolson) [reply src/valkey-cli.c]: That sounds fine to me.

Comment (zuiderkwast) [reply src/valkey-cli.c]: It's trivial to change it to just "UNBLOCKED" I believe, so no need to postpone it to a different PR? The strncmp below just checks that err starts with one of these strings if I read it correctly.

```suggestion
                "UNBLOCKED"};
```

**Review by madolson — APPROVED** [review]

**Pull request merged**